linode/manager
# M3-2767 fix: Show progress for Linode upgrades

## Description

This PR addresses a few different issues surrounding the Linode upgrade flow:

1) Previously, when upgrading a Linode from a legacy plan, no progress was displayed.
2) API has added new event types (`linode_mutate_create` and `linode_mutate_resize`) which we weren't handling.
3) When doing any sort of migration, the "Migration Imminent" notification didn't go away until after the migration was done.
4) The copy in the "Upgrade Available" banner was unclear.


**To fix <span> #</span>1 and <span>#</span>2:** Added the new event types, as well as a helper function, `removeBlacklistedEvents()` which filters out events we don't want to display. I chose to do this at the presentation level instead of Redux, because we might want to use these blacklisted events elsewhere. This means I also had to move the handling of `countUnseenEvents` to the UserEventMenu, because we don't want to show a badge for unseen events on the blacklist.

**To fix <span> #</span>3:** Added a new step in the events pipeline. Immediately after receiving events from the API, we check to see if we need to request notifications as well. This depends on the `action` of the most recent event, and the last time we updated notifications.

**To fix <span> #</span>4:** Removed the link to the resize page from the "Upgrade Available" banner.


## Type of Change
- Bug fix ('fix', 'repair', 'bug')
- Non breaking change ('update', 'change')

## Note to Reviewers

**To test:**
1) Have a Linode on a legacy plan (msg me for instructions)
2) Upgrade the Linode, or resize the Linode through the `Resize` tab.

I'd also play around with events in general and make sure everything is working as expected.
